The README.md was fetched from the upstream github repo at
https://raw.githubusercontent.com/anowell/pam-rs/refs/heads/master/README.md

diff --new-file -ur pam-bindings-0.1.1.orig/Cargo.toml pam-bindings-0.1.1/Cargo.toml
--- pam-bindings-0.1.1.orig/Cargo.toml	2025-05-22 12:53:15.635029396 +0100
+++ pam-bindings-0.1.1/Cargo.toml	2025-05-22 12:52:55.797752225 +0100
@@ -14,7 +14,7 @@
 version = "0.1.1"
 authors = ["Anthony Nowell <anowell@gmail.com>"]
 description = "PAM bindings for Rust"
-readme = "../README.md"
+readme = "README.md"
 keywords = [
     "pam",
     "ffi",
diff --new-file -ur pam-bindings-0.1.1.orig/README.md pam-bindings-0.1.1/README.md
--- pam-bindings-0.1.1.orig/README.md	1970-01-01 01:00:00.000000000 +0100
+++ pam-bindings-0.1.1/README.md	2025-05-22 12:52:48.020946435 +0100
@@ -0,0 +1,25 @@
+pam-rs
+========
+
+Rust interface to the pluggable authentication module framework (PAM).
+
+The goal of this library is to provide a type-safe API that can be used to
+interact with PAM.  The library is incomplete - currently it supports a subset
+of functions for use in a pam authentication module.  A pam module is a shared
+library that is invoked to authenticate a user, or to perform other functions.
+
+## üåê [pam-http](pam-http)
+
+An example of using pam-rs by performing HTTP basic access auth to authenticate users.
+
+## üçª [pam-sober](pam-sober)
+
+If you aren't sober enough for basic math, you can't login!
+
+### Credits
+
+The contents of this repo are heavily borrowed from:
+
+- [tozny/rust-pam](https://github.com/tozny/rust-pam)
+- [ndenev/pam_groupmap](https://github.com/ndenev/pam_groupmap)
+- [beatgammit/pam-http](https://github.com/beatgammit/pam-http)
